ecommerce/customers/models.py

```diff
@@ -4,10 +4,52 @@
 from django.core.mail import send_mail
 from django.utils import timezone
 from django.utils.translation import gettext_lazy as _
-
 from django.db import models
+from .managers import CustomerManager
+from django.core.validators import RegexValidator  # Import forms library to use forms.RegexField for Address.phone/postcode
+
+
+class Country(AbstractBaseUser):
+    """
+    returns: __str__(self)
+    params: BaseAbstractModel Class
+    """
+    name = models.CharField(verbose_name=_("name"), max_length=56)
+
+    def __srt__(self):
+        return self.name
+
 
-from customers.managers import CustomerManager
+class City(AbstractBaseUser):
+    """
+    returns: __str__(self)
+    params: BaseAbstractModel Class
+    """
+    name = models.CharField(verbose_name=_("name"), max_length=35)
+    country = models.ForeignKey(Country, on_delete=models.PROTECT)
+
+    def __str__(self):
+        return f"{self.name} - {self.country}"
+
+
+class Address(AbstractBaseUser):
```

Comment:
I think Country, City, and Address classes should inherit from **BaseAbstractModel** instead of AbstractBaseUser